ambry-api/src/main/java/com.github.ambry/account/Container.java

```diff
@@ -210,38 +266,95 @@ private Container(JSONObject metadata) throws JSONException {
     short metadataVersion = (short) metadata.getInt(JSON_VERSION_KEY);
     switch (metadataVersion) {
       case JSON_VERSION_1:
-        this.id = (short) metadata.getInt(CONTAINER_ID_KEY);
-        this.name = metadata.getString(CONTAINER_NAME_KEY);
-        this.status = ContainerStatus.valueOf(metadata.getString(STATUS_KEY));
-        this.description = metadata.optString(DESCRIPTION_KEY);
-        this.isPrivate = metadata.getBoolean(IS_PRIVATE_KEY);
-        this.parentAccountId = (short) metadata.getInt(PARENT_ACCOUNT_ID_KEY);
-        checkRequiredFields();
+        id = (short) metadata.getInt(CONTAINER_ID_KEY);
+        name = metadata.getString(CONTAINER_NAME_KEY);
+        status = ContainerStatus.valueOf(metadata.getString(STATUS_KEY));
+        description = metadata.optString(DESCRIPTION_KEY);
+        parentAccountId = (short) metadata.getInt(PARENT_ACCOUNT_ID_KEY);
+        boolean isPrivate = metadata.getBoolean(IS_PRIVATE_KEY);
+        // the V1 schema can optionally include the new V2 fields. If not present, some of these can be inferred from
+        // the isPrivate flag
+        encrypted = metadata.optBoolean(ENCRYPTED_KEY, false);
```

Comment:
sorry. this is confusing. If V1 is not expected to have these fields, then why look for it. why not directly assign defaults. I thought we decided not to have these optional fields in V1.